DevCycle/DevCycleClient.swift

```diff
@@ -406,23 +406,33 @@ public class DevCycleClient {
                         key: key,
                         value: nil,
                         defaultValue: defaultValue,
-                        evalReason: nil
+                        eval: EvalReason.defaultReason(details: "User Not Targeted")
                     )
                 }
 
                 self.variableInstanceDictonary[key]?.setObject(
                     variable, forKey: defaultValue as AnyObject)
             }
 
-            if !self.closed && !self.disableAutomaticEventLogging {
+            if !self.closed && !self.disableAutomaticEventLogging {                
                 self.eventQueue.updateAggregateEvents(
-                    variableKey: variable.key, variableIsDefaulted: variable.isDefaulted)
+                    variableKey: variable.key, variableIsDefaulted: variable.isDefaulted, metadata: createVariableEventMetaData(variableEval: variable.eval))
             }
 
             return variable
         }
     }
 
+    private func createVariableEventMetaData(variableEval: EvalReason?) -> [String: Any]? {
```

Comment:
can the return type here be more strictly typed?

Reply:
yup, should not be a problem